### DISCUSSION TITLE
MQTT and AMQP 1.0 server support, any plans?

### DISCUSSION THREAD

**Discussion by ppatierno** [smallrye/smallrye-reactive-messaging#2056] [Q&A] [answered]
Hi all,
AFAICS, Smallrye Reactive Messaging is mostly focused on the client side of messaging, because of the corresponding Eclipse Microprofile specification.
Referring to the MQTT and AMQP 1.0 protocols, it is using the corresponding Vert.x projects.
Such projects have support for the server part as well.
While in an MQTT or AMQP 1.0 messaging based architecture you want to use a full MQTT or AMQP 1.0 broker for exchanging messages, there are use cases where you are using a different platform (i.e. Apache Kafka) but you want to bridge from MQTT and AMQP 1.0.
I was wondering if there is any plan for Smallrye to support such a server part, useful for starting up an MQTT or AMQP 1.0 server for accepting connections and being able to bridge to other protocols.
It seems to me that something like an `MqttServerConnector` was available in the past (not sure about AMQP) but it's now gone.

**Comment by cescoffier** [comment, marked as answer]
Hello @ppatierno 

At the moment, we don't plan to investigate server components. It would be interesting, but at the same time, we need to limit the scope as we already have an ambitious and challenging roadmap.

**Reply by ppatierno** [reply]
Thanks for the reply @cescoffier ;-)
